client/src/context/useRequestContext.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState, createContext, FunctionComponent, useContext } from 'react';
+import { getBookings, sortBookings, updateBooking } from '../helpers/APICalls/bookingService';
+import { BookingApiData, BookingRequest } from '../interface/BookingApiData';
+import { useSnackBar } from './useSnackbarContext';
+
+export type Modify = 'Accept' | 'Decline';
+type ModifyBooking = (value: Modify, _id: string) => void;
+
+interface ReqContext {
+  bookings: BookingApiData;
+  fetching: boolean;
+  modifyBooking: ModifyBooking;
+}
+
+const RequestContext = createContext<ReqContext>({
+  bookings: {
+    upcoming: null,
+    current: [],
+    past: [],
+  },
+  fetching: true,
+  modifyBooking: () => null,
+});
+
+// cache the list of bookings for faster updates
+let bookingList: Array<BookingRequest> = [];
```

Comment:
Why not make `const` if it won't change?

Reply:
I didn't because I reassigned a new value to that variable in the following useffect hook